pkg/runtime/channels/channels.go

```diff
@@ -152,6 +161,81 @@ func (c *Channels) Refresh() error {
 	return nil
 }
 
+func (c *Channels) BuildHTTPPipelineFromComponents(comps []compsv1alpha1.Component, expectedPiplineType string) (middlehttp.Pipeline, error) {
```

Comment:
This should be moved to a common package so it can be used for the httpPipeline too

Reply:
Yeah , your right I have passed ```expectedPiplineType``` arg to support that .  And also adding making it stay in same package as channels is making more easy to use the utilities . Please let me know if I missed something